conn.go

```diff
@@ -73,14 +73,22 @@ func dial(ctx context.Context, addr string, num int, opt *Options) (*connect, er
 		}
 	}
 
-	compression := CompressionNone
+	var (
+		compression CompressionMethod
+		compressor  *compress.Writer
+	)
 	if opt.Compression != nil {
 		switch opt.Compression.Method {
-		case CompressionLZ4, CompressionZSTD, CompressionNone:
+		case CompressionLZ4, CompressionLZ4HC, CompressionZSTD, CompressionNone:
 			compression = opt.Compression.Method
 		default:
 			return nil, fmt.Errorf("unsupported compression method for native protocol")
 		}
+
+		compressor = compress.NewWriter(compress.Level(opt.Compression.Level), compress.Method(opt.Compression.Method))
+	} else {
+		compression = CompressionNone
```

Comment:
Is it really needed to initialize this given that the if in line https://github.com/ClickHouse/clickhouse-go/blob/main/conn.go#L247 always check for `!= CompressionNone`

I guess the point was to avoid having a nil compressor just in case?